internal/cephfs/nodeserver.go

```diff
@@ -127,13 +128,47 @@ func maybeUnlockFileEncryption(
 	stagingTargetPath string,
 	volID fsutil.VolumeID,
 ) error {
-	if volOptions.IsEncrypted() {
-		log.DebugLog(ctx, "cephfs: unlocking fscrypt on volume %q path %s", volID, stagingTargetPath)
 
+	if volOptions.IsEncrypted() == false {
+		return nil
+	}
+
+	// Define Mutex Lock variables
+	lockName := string(volID) + "-mutexLock"
+	lockCookie := lockName + "-coockie"
+	lockDesc := "Lock for " + string(volID)
+	lockDuration := 40 * time.Second
+	var flags byte = 0
+
+	log.DebugLog(ctx, "Creating lock for the following volume ID %s", volID)
+
+	ioctx, err := volOptions.GetConnection().GetIoctx(volOptions.MetadataPool)
+	if err != nil {
+	}
+	defer ioctx.Destroy()
+
+	res, err := ioctx.LockExclusive(volOptions.VolID, lockName, string(lockCookie), lockDesc, lockDuration, &flags)
```

Comment:
```suggestion
res, err := ioctx.LockExclusive(volOptions.VolID, lockName, lockCookie, lockDesc, lockDuration, &flags)
```
lockCookie is already a string